### ISSUE TITLE
getting image data from graphics context

### ISSUE THREAD

**Issue by mango314** [evancz/elm-graphics#10]
Here is an example I would like to work in Elm

```
    var width  = 960;
    var height = 500;

    var canvas = document.getElementById('vincent');
    var ctx = canvas.getContext("2d");
    canvas.width  = width;
    canvas.height = height;

    var image = new Image();
    image.src= "starry-night.jpg";
    image.onload = function () { 
        ctx.drawImage(image, 0, 0);
        image = ctx.getImageData(0, 0, width, height);
    }
```

[`Graphics`](http://package.elm-lang.org/packages/evancz/elm-graphics/1.0.0/) **has** a analogue of `ctx.drawImage(...)` now I am asking about getting the image data back out.

This is related to https://github.com/evancz/elm-graphics/issues/9

Can we call [`ctx.getImageData(...)`](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/getImageData) from Elm and get back a [`List Int`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Uint8ClampedArray) as the image data returns?   

Since I have experimented, I noticed it should be `List Color` or `List List Color` (a 2D array).  2D data might be stored as an underlying quadtree.  I don't know.

**Comment by mango314** [comment]
I'll say a little bit more.  Reading through [Collage](http://package.elm-lang.org/packages/evancz/elm-graphics/1.0.0/Collage) API, there are ways to add things **into** Collage, but not to query for information **from** collage.  

In particular, the color at a given pixel from 2D context.  [`ctx.getImageData()`](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/getImageData) is really the only query you can make in the 2D graphics context anyway.  Elm, in most cases, you have access to the original graphics themselves (which is always more precise) except for [`image`](http://package.elm-lang.org/packages/evancz/elm-graphics/1.0.0/Element#image)

---

Looking a bit through `Element.elm` I found the code for `image` based on a function that is not exposed:

```
image : Int -> Int -> String -> Element
image w h src =
    newElement w h (Image Plain w h src)
```

 `Image` is a type of `ElementPrim` and is just passed on to the Native JavaScript

```
type ElementPrim
    = Image ImageStyle Int Int String
    | Container RawPosition Element
    | Flow Direction (List Element)
    | Spacer
    | RawHtml
    | Custom
```

As far as I can tell from `.js` the Collage elements get stored as SVG until the very end and converted to canvas...

```
// IMAGES

function drawImage(redo, ctx, form)
{
    var img = new Image();
    img.onload = redo;
    img.src = form._3;
    var w = form._0,
        h = form._1,
        pos = form._2,
        srcX = pos._0,
        srcY = pos._1,
        srcW = w,
        srcH = h,
        destX = -w / 2,
        destY = -h / 2,
        destW = w,
        destH = h;

    ctx.scale(1, -1);
    ctx.drawImage(img, srcX, srcY, srcW, srcH, destX, destY, destW, destH);
}
```

In summary... I can never query the `ctx` because it's constantly being rebuilt.  I have some thinking to do... and certainly whatever solution I have won't be good for here.
